app/src/main/java/com/alexbar/layoutbasic/movies_app/utils/NotificationUtils.kt

```diff
@@ -0,0 +1,61 @@
+package com.alexbar.layoutbasic.movies_app.utils
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.TaskStackBuilder
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.net.toUri
+import com.alexbar.layoutbasic.MainActivity
+import com.alexbar.layoutbasic.R
+
+object NotificationUtils {
+    private const val CHANNEL_ID = "detail_new_movie_id"
+    private const val NOTIFICATION_ID = 1
+
+    fun showNotification(context: Context) {
+        val id = "12345"
+        val deepLinkIntent = Intent(
+            Intent.ACTION_VIEW,
+            "https://www.example.com/$id".toUri(),
+            context,
+            MainActivity::class.java
+        )
+        val deepLinkPendingIntent: PendingIntent? = TaskStackBuilder.create(context).run {
+            addNextIntentWithParentStack(deepLinkIntent)
+
+            val flag = if(Build.VERSION.SDK_INT > Build.VERSION_CODES.S) {
+                PendingIntent.FLAG_IMMUTABLE
+            }
+            else {
+                PendingIntent.FLAG_UPDATE_CURRENT
+            }
+            getPendingIntent(0, flag)
+        }
+
+        val notificationBuilder = NotificationCompat.Builder(context, CHANNEL_ID)
+            .setSmallIcon(R.drawable.play_icon)
+            .setContentTitle("New Movie Available")
+            .setContentText("Click here for more detail")
+            .setContentIntent(deepLinkPendingIntent)
+            .setAutoCancel(true)
+
+        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
```

Comment:
```
    private fun createNotificationChannel(context: Context) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channel = NotificationChannel(
                CHANNEL_ID,
                "Channel Name",
                NotificationManager.IMPORTANCE_DEFAULT
            ).apply {
                description = "Channel Description"
            }
            val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            notificationManager.createNotificationChannel(channel)
        }
    }
```

I've extracted the creation of the notification channel into a separate function createNotificationChannel. This improves readability and allows for better organization.